README.md
# plagiarism-checker README

The VS Code extension for plagiarism checker.

## Installation

Just download the ".vsix" file located in the "build" directory of this repository and install it in VS Code.

## Features

Created using VS Code Extension API and ReactJs as frontend.

## Usage Notes

Text should contain more than 15 words.

Text should contain less than 19000 characters.

**Enjoy!**

extension/CHANGELOG.md
# Change Log

All notable changes to the "plagiarism-checker" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

frontend/src/App.js
import { useState } from "react";
import "bootstrap/dist/css/bootstrap.css";
import { Badge, Button, Col, Container, Form, Row, Spinner } from "react-bootstrap";

function App() {
  const [checked, setChecked] = useState(false);
  const [uniquePercent, setUniquePercent] = useState(0);
  return (
    <>
      <script>{window.addEventListener("message", e => {
        if(e.data.percent) {
          setUniquePercent(e.data.percent);
          setChecked(true);
          document.getElementById("textError").hidden = true;
          document.getElementById("textError").innerText = "";
          document.getElementById("submitBtn").disabled = false;
          document.getElementById("submitBtnLoading").hidden = true;
        }
        else if(e.data.error) {
          setChecked(false);
          document.getElementById("textError").hidden = false;
          document.getElementById("textError").innerText = "An error occurred!";
          document.getElementById("submitBtn").disabled = false;
          document.getElementById("submitBtnLoading").hidden = true;
        }})}
      </script>
      <Container fluid="md">
        <Row>
          <Col style={{margin: "20px"}}><h1>Plagiarism Checker</h1></Col>
        </Row>
        <Row>
          <Form method="POST" id="form">
            <Col style={{margin: "20px"}}>
              <Form.Group>
                <Form.Label>Text</Form.Label>
                <Form.Control as="textarea" name="text" placeholder="Enter your text here" style={{height: "100px", backgroundColor: "#222", color: "#eee"}} />
                <Form.Text style={{color: "red"}} id="textError" hidden>abc</Form.Text>
              </Form.Group>
            </Col>
            <Col style={{margin: "20px"}}>
              <Form.Group>
                <Button type="submit" variant="outline-primary" id="submitBtn">Check plagiarism<Spinner id="submitBtnLoading" animation="border" variant="primary" size="sm" hidden /></Button>
              </Form.Group>
            </Col>
          </Form>
        </Row>
        {checked
        ? <Row className="text-center">
            <Col style={{margin: "20px"}} xs lg="3.5"></Col>
            <Col style={{margin: "20px"}} xs lg="3.5">
              <Row>
                <Col>Plagiarised</Col>
                <Col>Unique</Col>
              </Row>
              <Row>
                <Col><Badge bg="danger">{100 - uniquePercent}%</Badge></Col>
                <Col><Badge bg="success">{uniquePercent}%</Badge></Col>
              </Row>
            </Col>
            <Col style={{margin: "20px"}} xs lg="3.5"></Col>
          </Row>
          : <>
            </>
        }
      </Container>
    </>
  );
}

export default App;


extension/src/extension.ts
import * as vscode from "vscode";
import * as path from "path";
import axios from "axios";

export function activate(context: vscode.ExtensionContext) {
	let webpanel: vscode.WebviewPanel | undefined;
	let disposable = vscode.commands.registerCommand("plagiarism-checker.openWindow", () => {
		if(!webpanel) {
			webpanel = vscode.window.createWebviewPanel("plagiarism-checker.window", "Plagiarism Checker", vscode.ViewColumn.Active, {enableScripts: true, localResourceRoots: [vscode.Uri.file(path.join(context.extensionPath, "ui"))]});
			let loadPanel = function() {
				if(webpanel) {
					let faviconUri = webpanel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, "ui", "favicon.ico")));
					let manifestUri = webpanel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, "ui", "manifest.json")));
					let jsUri = webpanel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, "ui/static/js", "main.ec2ef2c0.js")));
					let cssUri = webpanel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, "ui/static/css", "main.ababef68.css")));
					webpanel.webview.html = `
					<!doctype html>
						<html lang="en">
						
						<head>
							<meta charset="utf-8" />
							<link rel="icon" href="${faviconUri}" />
							<meta name="viewport" content="width=device-width,initial-scale=1" />
							<meta name="theme-color" content="#000000" />
							<meta name="Plagiarism-Checker" content="Web site created for checking plagiarism." />
							<link rel="manifest" href="${manifestUri}" />
							<title>Plagiarism Checker</title>
							<script defer="defer" src="${jsUri}"></script>
							<link href="${cssUri}" rel="stylesheet">
						</head>
					
						<body style="background-color: #222; color: #eeeeee9f;">
							<noscript>You need to enable JavaScript to run this app.</noscript>
							<script>
								setTimeout(function(){
									const vscodeApi = acquireVsCodeApi();
									const handleSubmit = (e) => {
										e.preventDefault();
										const formData = new FormData(e.target);
										document.getElementById("submitBtn").disabled = true;
          								document.getElementById("submitBtnLoading").hidden = false;
										if(formData.get("text").split(" ").length > 15) {
											if(formData.get("text").split("").length <= 19000) {
												vscodeApi.postMessage({
													text: formData.get("text")
												});
											}
											else {
												document.getElementById("textError").hidden = false;
												document.getElementById("textError").innerText = "Maximum 19000 characters are allowed.";
											}
										}
										else {
											document.getElementById("textError").hidden = false;
											document.getElementById("textError").innerText = "Please enter more than 15 words to check.";
										}
									}
									document.getElementById("form").onsubmit = handleSubmit;
								}, 1000);
							</script>
							<div id="root"></div>
						</body>
					
					</html>`;
				}
			}
			webpanel.onDidDispose(() => {
				webpanel = undefined;
		  	}, null, context.subscriptions);
			webpanel.webview.onDidReceiveMessage(e => {
				axios.post("https://papersowl.com/plagiarism-checker-send-data", {
					is_free: false,
					plagchecker_locale: "en",
					title: "",
					text: e.text.replace(" ", "+")
				}, {
					"headers": {
						"accept": "*/*",
						"content-type": "application/x-www-form-urlencoded; charset=UTF-8",
						"sec-ch-ua": "\"Not?A_Brand\";v=\"8\", \"Chromium\";v=\"108\", \"Google Chrome\";v=\"108\"",
						"sec-ch-ua-mobile": "?0",
						"sec-ch-ua-platform": "\"Windows\"",
						"x-requested-with": "XMLHttpRequest",
						"Referer": "https://papersowl.com/free-plagiarism-checker",
						"Referrer-Policy": "strict-origin-when-cross-origin"
					}
				}).then((res: any) => {
					webpanel?.webview.postMessage({percent: res.data.percent});
				}).catch((err: any) => {console.error(err); webpanel?.webview.postMessage({error: true});});
		})
			loadPanel();
		}
		else {
			webpanel.reveal();
		}
	});
	context.subscriptions.push(disposable);
}

export function deactivate() {}


extension/out/extension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deactivate = exports.activate = void 0;
const vscode = require("vscode");
const path = require("path");
const axios_1 = require("axios");
function activate(context) {
    let webpanel;
    let disposable = vscode.commands.registerCommand("plagiarism-checker.openWindow", () => {
        if (!webpanel) {
            webpanel = vscode.window.createWebviewPanel("plagiarism-checker.window", "Plagiarism Checker", vscode.ViewColumn.Active, { enableScripts: true, localResourceRoots: [vscode.Uri.file(path.join(context.extensionPath, "ui"))] });
            let loadPanel = function () {
                if (webpanel) {
                    let faviconUri = webpanel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, "ui", "favicon.ico")));
                    let manifestUri = webpanel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, "ui", "manifest.json")));
                    let jsUri = webpanel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, "ui/static/js", "main.ec2ef2c0.js")));
                    let cssUri = webpanel.webview.asWebviewUri(vscode.Uri.file(path.join(context.extensionPath, "ui/static/css", "main.ababef68.css")));
                    webpanel.webview.html = `
					<!doctype html>
						<html lang="en">
						
						<head>
							<meta charset="utf-8" />
							<link rel="icon" href="${faviconUri}" />
							<meta name="viewport" content="width=device-width,initial-scale=1" />
							<meta name="theme-color" content="#000000" />
							<meta name="Plagiarism-Checker" content="Web site created for checking plagiarism." />
							<link rel="manifest" href="${manifestUri}" />
							<title>Plagiarism Checker</title>
							<script defer="defer" src="${jsUri}"></script>
							<link href="${cssUri}" rel="stylesheet">
						</head>
					
						<body style="background-color: #222; color: #eeeeee9f;">
							<noscript>You need to enable JavaScript to run this app.</noscript>
							<script>
								setTimeout(function(){
									const vscodeApi = acquireVsCodeApi();
									const handleSubmit = (e) => {
										e.preventDefault();
										const formData = new FormData(e.target);
										document.getElementById("submitBtn").disabled = true;
          								document.getElementById("submitBtnLoading").hidden = false;
										if(formData.get("text").split(" ").length > 15) {
											if(formData.get("text").split("").length <= 19000) {
												vscodeApi.postMessage({
													text: formData.get("text")
												});
											}
											else {
												document.getElementById("textError").hidden = false;
												document.getElementById("textError").innerText = "Maximum 19000 characters are allowed.";
											}
										}
										else {
											document.getElementById("textError").hidden = false;
											document.getElementById("textError").innerText = "Please enter more than 15 words to check.";
										}
									}
									document.getElementById("form").onsubmit = handleSubmit;
								}, 1000);
							</script>
							<div id="root"></div>
						</body>
					
					</html>`;
                }
            };
            webpanel.onDidDispose(() => {
                webpanel = undefined;
            }, null, context.subscriptions);
            webpanel.webview.onDidReceiveMessage(e => {
                axios_1.default.post("https://papersowl.com/plagiarism-checker-send-data", {
                    is_free: false,
                    plagchecker_locale: "en",
                    title: "",
                    text: e.text.replace(" ", "+")
                }, {
                    "headers": {
                        "accept": "*/*",
                        "content-type": "application/x-www-form-urlencoded; charset=UTF-8",
                        "sec-ch-ua": "\"Not?A_Brand\";v=\"8\", \"Chromium\";v=\"108\", \"Google Chrome\";v=\"108\"",
                        "sec-ch-ua-mobile": "?0",
                        "sec-ch-ua-platform": "\"Windows\"",
                        "x-requested-with": "XMLHttpRequest",
                        "Referer": "https://papersowl.com/free-plagiarism-checker",
                        "Referrer-Policy": "strict-origin-when-cross-origin"
                    }
                }).then((res) => {
                    webpanel?.webview.postMessage({ percent: res.data.percent });
                }).catch((err) => { console.error(err); webpanel?.webview.postMessage({ error: true }); });
            });
            loadPanel();
        }
        else {
            webpanel.reveal();
        }
    });
    context.subscriptions.push(disposable);
}
exports.activate = activate;
function deactivate() { }
exports.deactivate = deactivate;
//# sourceMappingURL=extension.js.map